### ISSUE TITLE
[$250] Task – Task disappears from LHN when it is not open

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#46715]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.16-0
**Reproducible in staging?:** Y
**Reproducible in production?:** Y 
**If this was caught during regression testing, add the test name, ID and link from TestRail:** https://expensify.testrail.io/index.php?/tests/view/4804049
**Email or phone of affected tester (no customers):** ponikarchuks+32824@gmail.com
**Issue reported by:** Applause - Internal Team

## Action Performed:
1. Go to https://staging.new.expensify.com/
2. Log in
3. Open chat with user B
4. Create a task without assignee
5. Open task details and assign it to you
6. Observe it in LHN
7. Open another chat and open task details in LHN
8. Open another chat and observe LHN

## Expected Result:
Task present in LHN

## Actual Result:
Task disappears from LHN

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [x] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/user-attachments/assets/416b4d08-8df6-4e50-941f-73430e46d46b

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~015e6ab873971b6bf9</li>
        <li>Upwork Job ID: 1820503061150367371</li>
        <li>Last Price Increase: 2024-08-26</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @hungvu193</details>

**Comment by lanitochka17** [comment]
We think that this bug might be related to #vip-vsp

**Comment by FitseTLT** [comment]
BE.
The reason it appears in LHN in (6) is because the task report has become unread because when you assign yourself as an assigne the `lastMentionedTime` will be updated, but then it doesn't appear in step 8 because we have opened the task report and it is no more unread report.
The only thing that should be fixed here that lastMentionedTime should not be updated if a person assign himself for a task it only makes sense to assume it as a mention if a person is assigned by another user but that, I have checked, it is a BE issue can only be fixed in BE.

**Comment by twisterdotcom** [comment]
Sounds good @FitseTLT, gonna get a C+ to confirm.

**Comment by hungvu193** [comment]
Daily update: I'm quite busy today but definitely can review this one tomorrow morning

**Comment by hungvu193** [comment]
> 2\. After the above fix, we can notice another bug, related to the GBR.

May I know what's the GBR bug that you mentioned?

**Comment by hungvu193** [comment]
> > May I know what's the GBR bug that you mentioned?
> 
> In step 5, In offline mode, we don't have the GBR when we assign the task ourselves. But in online mode, the GBR is displayed.

Thanks, I think that's good to fix but not related to this issue 🤔 .

My two cents here, I think a task shouldn't disappear from LHN if it's a self-assigned task (We fixed a similar bug [here](https://github.com/Expensify/App/issues/45543)). We can fix this from BE or update FE's `shouldOverrideHidden` logic.

**Comment by hungvu193** [comment]
> I think a task shouldn't disappear from LHN if it's a self-assigned task (We fixed a similar bug [here](https://github.com/Expensify/App/issues/45543)).

@twisterdotcom How do you think about this one?

**Comment by puneetlath** [comment]
This isn't a bug. Tasks are like expense reports. They don't show in your LHN if you have access to the parent.

**Comment by hungvu193** [comment]
> > This isn't a bug. Tasks are like expense reports. They don't show in your LHN if you have access to the parent.
> 
> @hungvu193 if it is expected, so there is a bug when self-assigning as task when offline. In there, the task report is displayed in LHN.

@puneetlath  What do you think about this case?

**Comment by twisterdotcom** [comment]
I agree here:

>They don't show in your LHN if you have access to the parent.

But the bug shown in the video is that it is pinned with a GBR after creation and when navigating away, the task is removed from the LHN, even though it had a GBR. It's a confusing flow to show the GBR at both the highest and lowest context at once in the LHN because it feels like it shouldn't disappear.

**Comment by twisterdotcom** [comment]
I agree here:

>They don't show in your LHN if you have access to the parent.

But the bug shown in the video is that it is pinned with a GBR after creation and when navigating away, the task is removed from the LHN, even though it had a GBR. It's a confusing flow to show the GBR at both the highest and lowest context at once in the LHN because it feels like it shouldn't disappear.

**Comment by puneetlath** [comment]
Ahh I see, you're right. Ignore me. That does seem to be a bug.

**Comment by hungvu193** [comment]
Thanks for confirming.
Waiting for proposals

**Comment by hungvu193** [comment]
> App/src/libs/actions/Task.ts

I don't think that's the correct solution, remove `Hidden` as `notificationPreference` will make the task visible even when it doesn't have GBR. We still want to keep the task hidden if user has access to its parent and only show it if it has GBR.

**Comment by hungvu193** [comment]
Thanks for your update! I'll take a look today

**Comment by hungvu193** [comment]
> > Wait, is there any case that user cannot access the parent report of the task?
> 
> For example, you can open the selfDM, then create a task then assign it to anyone (named user B). Then you sign in with user B. User B cannot access the selfDM (the parent report of the task).

That's the user that got assigned to the task, if you create te task you should always access the parent report. Then this should be fixed from our BE.

**Comment by hungvu193** [comment]
I mean this issue should be fixed from our BE.

**Comment by hungvu193** [comment]
> Do you think we need to fix the issue in FE, in other words, the optimistic data. Let's take a look at the code below:
> 
> https://github.com/Expensify/App/blob/f7e265a65f695edd85870bd0c8048cf4578e14fb/src/libs/actions/Task.ts#L584-L586
> 
> Now if you assign the task to yourself, we always set the `notificationPreference` to `always` (leads to the task report is displayed in LHN), regardless of whether the user can access the parent report or not, right?

That makes sense, however there's another issue, when you self-assign task, you see a GBR, a task with GBR shouldn't be hidden.

**Comment by twisterdotcom** [comment]
I do not truly understand the data hierarchy entirely, but I think if you have a GBR, and we always show the highest context for that GBR, we should show whatever report is the highest context for that GBR, even if you may not have access to the absolute highest - does that make sense?

ie, you can create a private room and then allocate tasks to individuals outside of that room. Just because that individual isn't a member of that room, if their is a task waiting on them, it should show in their LHN.

**Comment by hungvu193** [comment]
> I do not truly understand the data hierarchy entirely, but I think if you have a GBR, and we always show the highest context for that GBR, we should show whatever report is the highest context for that GBR, even if you may not have access to the absolute highest - does that make sense?

I also agree with this one.

**Comment by hungvu193** [comment]
@daledah  No need to update.
Sorry for the delay, I've just tested again and @daledah 's alternative solution looks good to me! 
We're currently waiting for the BE data to hide the child task, meanwhile if we can access the parent task, the notification should be hidden by default.

🎀 👀 🎀  C+ reviewed

**Comment by hungvu193** [comment]
Waiting for assigning

**Comment by twisterdotcom** [comment]
Bump @aldo-expensify

**Comment by aldo-expensify** [comment]
> @aldo-expensify I've drafted a PR to fix the FE side. Waiting for BE fix.

Thanks @daledah , after reading the whole thread, I'm still a bit confused of what is the expected behaviour and what are the backend changes we want to do.

From testing, I see that when I open the unassigned task, it is added to the LHN. Then, I assign the task to myself and then I move away to another report. The newly assigned task remains in the LHN when it should have been removed... this is what the bug is about, right?

The proposed backend change is that `lastMentionedTime` shouldn't be updated when the task is self assigned, correct?

**Comment by hungvu193** [comment]
@daledah Let's complete and move forward with your PR

**Comment by hungvu193** [comment]
@twisterdotcom  This PR was deployed to prod 2 weeks ago. This issue is now ready for payment

**Comment by twisterdotcom** [comment]
Payment Summary:

- @hungvu193 paid $250 via manual requests (C+)
- @daledah paid $250 via Upwork [here](https://www.upwork.com/nx/wm/offer/new/1062156562) (Contributor)

**Comment by garrettmknight** [comment]
$250 approved for @hungvu193
